multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -718,14 +709,28 @@ def fit(
         if self._config is not None:  # continuous training
             config = self._config
 
-        problem_type, output_shape = infer_problem_type_output_shape(
+        # FIXME: Align logic with Tabular,
+        #  don't combine output_shape and problem_type detection, make them separate
```

Comment:
Agree. We need to separate the logic. The data split inconsistency between two fit() is due to that we tie them together and don't infer problem type before splitting data.